quic/s2n-quic-core/src/recovery/hybrid_slow_start.rs

```diff
@@ -94,24 +137,56 @@ impl HybridSlowStart {
         if let (N_SAMPLING, Some(last_min_rtt), Some(cur_min_rtt)) =
             (self.sample_count, self.last_min_rtt, self.cur_min_rtt)
         {
-            let threshold = last_min_rtt / THRESHOLD_DIVIDEND;
-            // Clamp n to the min and max thresholds
-            let threshold = threshold.min(MAX_DELAY_THRESHOLD).max(MIN_DELAY_THRESHOLD);
-            let delay_increase_is_over_threshold = cur_min_rtt >= last_min_rtt + threshold;
-            let congestion_window_is_above_minimum = congestion_window >= self.low_ssthresh();
-
-            if delay_increase_is_over_threshold && congestion_window_is_above_minimum {
-                self.threshold = congestion_window;
+            if congestion_window >= self.css_threshold {
+                self.css_count += 1;
+                if cur_min_rtt < self.css_baseline_min_rtt {
+                    // resume slow start
+                    self.css_threshold = self.threshold;
+                    self.ss_growth_divisor = 1.0;
+                    self.css_count = 0;
+                }
+                if self.css_count >= CSS_ROUNDS {
+                    // exit slow start phase
+                    self.threshold = congestion_window;
+                    self.css_threshold = f32::MAX;
+                    self.ss_growth_divisor = 1.0;
+                }
```

Comment:
Could you make it a little clearer what parts of this are hystart++, maybe put the hystart++ code in a separate method? Right not it isn't very obvious unless you remember `self.css_threshold` will always be f32:MAX is its not hystart++. So this could also check `self.use_hystart_plus_plus`